### ISSUE TITLE
비콘 감지시 백그라운드 작업 관련 문의 드립니다.

### ISSUE THREAD

**Issue by tryzone** [wizTurn/Android-SDK#58]
안녕하세요.

현재 wizturn 비콘을 이용하여 앱을 개발중에 있습니다.

다름이 아니라 앱에서 비콘을 감지하는 서비스를 돌리면 앱의 백그라운드 작업이 동작 하지 않는 

문제가 있습니다. 이 문제에 관하여 인지하고 계신지, 혹은 해결 방법이 있는지 문의 드립니다.

참고로 앱에서 하는 백그라운드 작업은 비동기 통신(AsyncTask)입니다.

해당 문제(비동기 통신)는 비콘 감지를 끄면 작동을 하고, 비콘 감지를 스타트 하면 작동하지 않습니다.

이에 대한 답변 부탁드립니다.

**Comment by jayjay-dio** [comment]
안녕하세요. 안드로이드 개발 담당자입니다.
말씀하신 이슈는 아직 제가 파악되지 않는 문제입니다. AsyncTask, Thread, Executors등을 이용한 백그라운드 처리시 WIZTURN SDK의 메소드를 호출은 UI Thread에서 호출되어야 합니다.
AsyncTask의 경우 onPostExecute()에서 WIZTURN SDK의 메소드를 호출하시면 문제가 없을 것 같습니다. 
Android의 경우 Bluetooth API호출은 UI(Main) Thread에서 호출하는것을 권장하고 있습니다.

상기와 같이 처리하셨는데 문제가 발생하시면 답변 주시면 감사하겠습니다.

**Comment by tryzone** [comment]
빠른 답변 감사합니다. 비콘 감지를 서비스로 돌리는 이유는 사용자가 앱을 종료 하였을때에도 지속적으로 감지 하기 위함입니다. 메인스레드에서 비콘감지를 스타트 한 후 앱을 종료해도 비콘 감지가 가능한지요.

**Comment by tryzone** [comment]
비콘 감지를 메인 쓰레드에서 호출해도 앱 종료후에 감지가 되는 것을 확인 하였습니다.

하지만 비동기 통신 문제는 해결되지 않네요.

**Comment by jayjay-dio** [comment]
네,, AsyncTask에서 CentralManager.startScanning()을 했는데 비콘 스캐닝이 멈췄고, CentralManager.stopScanning()을 했는데 스캔을 시작했다라는 말씀이신가요? 다시 말하면 AsyncTask에서 비콘 스캔이 반대로 작동한다는건가요?
제가 문제가 무엇인지 정확히 알기위해 자세히 물어봤습니다~~.

**Comment by tryzone** [comment]
CentralManager 는 무엇인가요?

일단 비콘 스캐닝 스타트는 다음과 같이 했습니다.
WizTurnManager.sharedInstance(context).setInitController();				WizTurnManager.sharedInstance(context).setWizTurnDelegate(_wtDelegate);				WizTurnManager.sharedInstance(context).startController();

위에 보이는 WizTUrnManager가 싱글턴으로 구현되어있는거 같은데 CentralManager는 다른 것인지요.


증상은 간단히 말씀드리면
'비콘 스캐닝이 시작되면 비동기 통신이 작동하지 않는다' 라는 것 입니다.

**Comment by jayjay-dio** [comment]
옛날 SDK를 쓰시도 계시는것 같습니다. 최근 버전 2.0.0을 쓰시길 권장드립니다.

**Comment by tryzone** [comment]
넵.. 친절한 답변 감사합니다. 
제가 옛날 SDK를 쓰고있었나보네요 ㅠㅠ
SDK 교체 후 다시 작업 해 보도록 하겠습니다.

다시 한 번 답변에 감사드립니다.
